PSScripts/Get-VersionedFunctionAppName.ps1

```diff
@@ -0,0 +1,127 @@
+<#
+    .SYNOPSIS
+    Get a function app name and version from it's source branch
+
+    .DESCRIPTION
+    Gets the function app name and version from it's branch name and it's base name
+
+    .EXAMPLE
+    Get-VersionedFunctionAppName.ps1 -BranchName someBranch -FunctionBaseName some-function-app
+
+    .PARAMETER BranchName
+    The name of a branch
+
+    .PARAMETER FunctionAppBaseName
+    The function app name minus the version, eg dss-at-cust-fa rather than dss-at-cust-v2-fa
+#>
+
+param()
+
+function Write-AzureDevopsVariable {
+    <#
+        .SYNOPSIS
+        Writes a string representing azure devops variable to the output
+
+        .EXAMPLE
+        Write-AzureDevopsVariable -Name SomeVariable -Value SomeValue
+
+        .PARAMETER Name
+        The name of the variable
+
+        .PARAMETER Value
+        The value of the variable
+
+        .PARAMETER IsOuput
+        Determines if the value should be propogated to other tasks steps.
+        Optional, defaults to false.
+    #>
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $Name,
+        [Parameter(Mandatory=$true)]
+        [string] $Value,
+        [bool] $IsOutput = $false
+    )
+
+    $lowerCaseOutputValue = ([string] $IsOutput).ToLower()
+
+    Write-Output "##vso[task.setvariable variable=$Name;isOutput=$lowerCaseOutputValue]$Value"
+}
+
+
+function Get-FunctionVersionFromBranch {
+    <#
+        .SYNOPSIS
+        Get the version of a function from it's branch name
+
+        .EXAMPLE
+        Get-VersionedFunctionAppNameFromBranch -BranchName someBranch
+
+        .PARAMETER BranchName
+        The name of a branch
+    #>
+
+    param(
+        [Parameter(Mandatory=$true)]
+        [string] $BranchName
+    )
+
+    if($BranchName -iin @("dev", "master"))  {
+        $branchVersion = "v1"
+    } else {
+        if($BranchName -match '^(?:(v[0-9]*)-)?(?:master|dev)(?:-(v[0-9]*))?$') {
```

Comment:
I think you used + rather than * in the regexes in other scripts that are checking the branch name.  Should they be consistent or are they trying to achieve different aims?